KAFKA-9863: update the deprecated --zookeeper option in the documentation into --bootstrap-server

showuon opened the pull request:
Since V2.2.0, the -zookeeper option turned into deprecated because Kafka can directly connect to brokers with --bootstrap-server (KIP-377). But in the official documentation, there are many example commands use --zookeeper instead of --bootstrap-server. Follow the command in the documentation, you'll get this warning, which is not good.

```
Warning: --zookeeper is deprecated and will be removed in a future version of Kafka.
Use --bootstrap-server instead to specify a broker to connect to.
```

Update configuration.html, ops.html, security.html files

Mainly update 2 commands:
1. before: `bin/kafka-configs.sh --zookeeper localhost:2181 ...`
    after: `bin/kafka-configs.sh --bootstrap-server localhost:9092 ...`
2. before: `bin/kafka-reassign-partitions.sh --zookeeper localhost:2181 ...`
    after: `bin/kafka-reassign-partitions.sh --bootstrap-server localhost:9092 ...`

JIRA: https://issues.apache.org/jira/browse/KAFKA-9863

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

showuon committed:
update the deprecated --zookeeper option in the documentation into --bootstrap-server

showuon commented:
@joestein @rajinisivaram @hachikuji , could you help review this PR? Thank you very much.

rondagostino commented on docs/configuration.html:
```diff
@@ -100,7 +100,7 @@ <h5>Updating Password Configs in ZooKeeper Before Starting Brokers</h5>
 
   <pre class="brush: bash;">
-  &gt; bin/kafka-configs.sh --zookeeper localhost:2181 --entity-type brokers --entity-name 0 --alter --add-config
+  &gt; bin/kafka-configs.sh --bootstrap-server localhost:9092 --entity-type brokers --entity-name 0 --alter --add-config
```
This specific change should not be made because the paragraph above is referring specifically to configuration "before starting brokers for bootstrapping".  However, since it is talking about storing sensitive information, it is likely that this would be a TLS-enabled Zookeeper quorum.  So I would suggest:
```suggestion
  &gt; bin/kafka-configs.sh --zookeeper localhost:2182 --zk-tls-config-file zk_tls_config.properties --entity-type brokers --entity-name 0 --alter --add-config
```

showuon replied:
Updated. Thanks.

rondagostino commented on docs/security.html:
```diff
@@ -754,22 +754,22 @@ <h3><a id="security_sasl" href="#security_sasl">7.3 Authentication using SASL</a
         <p>Create SCRAM credentials for user <i>alice</i> with password <i>alice-secret</i>:
         <pre class="brush: bash;">
-    > bin/kafka-configs.sh --zookeeper localhost:2181 --alter --add-config 'SCRAM-SHA-256=[iterations=8192,password=alice-secret],SCRAM-SHA-512=[password=alice-secret]' --entity-type users --entity-name alice
+    > bin/kafka-configs.sh --bootstrap-server localhost:9092 --alter --add-config 'SCRAM-SHA-256=[iterations=8192,password=alice-secret],SCRAM-SHA-512=[password=alice-secret]' --entity-type users --entity-name alice
```
Same thing here and the examples below: this is about bootstrapping credentials.
```suggestion
    > bin/kafka-configs.sh --zookeeper localhost:2182 --zk-tls-config-file zk_tls_config.properties --alter --add-config 'SCRAM-SHA-256=[iterations=8192,password=alice-secret],SCRAM-SHA-512=[password=alice-secret]' --entity-type users --entity-name alice
```

showuon replied:
Updated. Thanks.

rondagostino commented on docs/security.html:
```diff
@@ -754,22 +754,22 @@ <h3><a id="security_sasl" href="#security_sasl">7.3 Authentication using SASL</a
         <p>The following examples also require a user <i>admin</i> for inter-broker communication which can be created using:
         <pre class="brush: bash;">
-    > bin/kafka-configs.sh --zookeeper localhost:2181 --alter --add-config 'SCRAM-SHA-256=[password=admin-secret],SCRAM-SHA-512=[password=admin-secret]' --entity-type users --entity-name admin
+    > bin/kafka-configs.sh --bootstrap-server localhost:9092 --alter --add-config 'SCRAM-SHA-256=[password=admin-secret],SCRAM-SHA-512=[password=admin-secret]' --entity-type users --entity-name admin
```
And here...

rondagostino commented on docs/security.html:
```diff
@@ -754,22 +754,22 @@ <h3><a id="security_sasl" href="#security_sasl">7.3 Authentication using SASL</a
         <p>Existing credentials may be listed using the <i>--describe</i> option:
         <pre class="brush: bash;">
-    > bin/kafka-configs.sh --zookeeper localhost:2181 --describe --entity-type users --entity-name alice
+    > bin/kafka-configs.sh --bootstrap-server localhost:9092 --describe --entity-type users --entity-name alice
```
And here...

showuon replied:
Updated. Thanks.

rondagostino commented on docs/security.html:
```diff
@@ -754,22 +754,22 @@ <h3><a id="security_sasl" href="#security_sasl">7.3 Authentication using SASL</a
         <p>Credentials may be deleted for one or more SCRAM mechanisms using the <i>--delete</i> option:
         <pre class="brush: bash;">
-    > bin/kafka-configs.sh --zookeeper localhost:2181 --alter --delete-config 'SCRAM-SHA-512' --entity-type users --entity-name alice
+    > bin/kafka-configs.sh --bootstrap-server localhost:9092 --alter --delete-config 'SCRAM-SHA-512' --entity-type users --entity-name alice
```
...and here.

showuon replied:
Updated. Thanks.

rondagostino reviewed (commented):
Thanks for the PR!  This is an excellent change.  There is one other occurrences that should also be fixed.  From `git grep -n -- --zookeeper *.html`:

`docs/ops.html:97:  &gt; bin/kafka-preferred-replica-election.sh --zookeeper zk_host:port/chroot`

showuon committed:
address reviewer's comments

1. keep --zookeeper argument in the places that is talking about configure zk before broker up
2. use --zookeeper --zk-tls-config-file when doing some security related action, ex: update password

showuon commented:
Hi @rondagostino , thanks for the comments. 

What I've updated in https://github.com/apache/kafka/pull/8482/commits/bcc815508e75b8b9ed9b277e9e9d937d0ea1ca29:
1. Yes, I missed the place in `docs/ops.html:97: &gt; bin/kafka-preferred-replica-election.sh --zookeeper zk_host:port/chroot`, I updated it to `&gt; bin/kafka-preferred-replica-election.sh --bootstrap-server broker_host:port` now. 
2. Keep using --zookeeper in the place that talking about configuration before broker up. 
3. Use secure port 2182 instead of 2181, and add `--zk-tls-config-file` to configure the tls connection with zookeeper when doing security-related actions.

BTW, enable the Kafka client to use TLS connection with zookeeper is a really good KIP. Thank you very much.

rondagostino reviewed (commented):
> BTW, enable the Kafka client to use TLS connection with zookeeper is a really good KIP. Thank you very much.

You're welcome , @showuon -- thank you for these fixes.  LGTM.

showuon commented:
@cmccabe , could you please help review this PR? 
This is a small and straightforward change to update the documentation only. And it's related to the PR that you also reviewed before https://github.com/apache/kafka/pull/8056 , and https://github.com/apache/kafka/pull/3848

Thank you very much.

showuon commented:
@junrao @omkreddy , could you please help review this PR? This is a small and straightforward change to update the documentation only, but I think it's important to users. Thank you very much.

cmccabe commented:
LGTM.  Thanks, @showuon , @rondagostino

gerardq commented:
Hi, don't know if this is the correct place but here is my issue: `kafka-configs --zookeeper ... ---entity-type users` can be used to manage SCRAM users stored in zookeeper. So it still needs to connect to zookeeper, not bootstrap server. Will this be supported in future releases? Whenever I run the command (using cp 5.5.0) I am getting the warning '--zookeeper is deprecated and will be removed in a future version of Kafka'. When trying the command via bootstrap-server it fails saying 'Invalid entity type users, the entity type must be one of topics,brokers,broker-loggers with the --bootstrap-server argument'.

cmccabe commented:
@gerardq : https://cwiki.apache.org/confluence/display/KAFKA/KIP-506%3A+Allow+setting+SCRAM+password+via+Admin+interface will address this in the future.
